src/kontrol/solc_to_k.py

```diff
@@ -1295,3 +1274,24 @@ def _find_function_calls(node: dict) -> None:
 
     _find_function_calls(node)
     return function_calls
+
+
+def process_storage_layout(storage_layout: dict) -> tuple[StorageField, ...]:
+    storage = storage_layout.get('storage', [])
+    types = storage_layout.get('types', {})
+
+    fields_list: list[StorageField] = []
+    for field in storage:
+        try:
+            type_info = types.get(field['type'], {})
+            storage_field = StorageField(
+                label=field['label'],
+                data_type=type_info.get('label', field['type']),
+                slot=int(field['slot']),
+                offset=int(field['offset']),
+            )
+            fields_list.append(storage_field)
+        except (KeyError, ValueError) as e:
+            _LOGGER.error(f'Error processing field {field}: {e}')
```

Comment:
Is it OK for a field to be not parsed in case of failure? If not, an exception should be raised here.

Reply:
I'm not sure at this moment what would cause a failure like that. I can raise an exception here instead in the next PR.